annotationPipeline/src/main/java/org/cbioportal/annotation/pipeline/MutationRecordReader.java

```diff
@@ -0,0 +1,157 @@
+    private List<AnnotatedRecord> annotatedRecords = new ArrayList<>();
+    
+    @Autowired
+    Annotator annotator;
+    @Autowired
+    BatchConfiguration config;
+    
+    @Override
+    public void open(ExecutionContext ec) throws ItemStreamException {
+        
+        processComments();
+        
+        FlatFileItemReader<MutationRecord> reader = new FlatFileItemReader<>();
+        reader.setResource(new FileSystemResource(filename));
+        DefaultLineMapper<MutationRecord> mapper = new DefaultLineMapper<>();
+        final DelimitedLineTokenizer tokenizer = new DelimitedLineTokenizer();
+        tokenizer.setDelimiter("\t");
+        mapper.setLineTokenizer(tokenizer);
+        mapper.setFieldSetMapper(new MutationFieldSetMapper());
+        reader.setLineMapper(mapper);
+        reader.setLinesToSkip(1);
+        reader.setSkippedLinesCallback(new LineCallbackHandler() {
+                @Override
+                public void handleLine(String line) {
+                    tokenizer.setNames(line.split("\t"));
+                }
+        });
+        
+        reader.open(ec);
+        
+        MutationRecord mutationRecord;
+        
+        try {
+            while((mutationRecord = reader.read()) != null) {
+                mutationRecords.add(mutationRecord);
+            }
+        }
+        catch(Exception e) {
+            throw new ItemStreamException(e);
+        }
+        
+        for(MutationRecord record : mutationRecords) {
+            annotatedRecords.add(annotator.annotateRecord(record, replace.equals("true")));
+        }
+    }
+
+    @Override
+    public void update(ExecutionContext ec) throws ItemStreamException {}
+
+    @Override
+    public void close() throws ItemStreamException {}
+
+    @Override
+    public AnnotatedRecord read() throws Exception {
+        if (!annotatedRecords.isEmpty()) {            
+            return annotatedRecords.remove(0);         
+        }
+        return null;
+    }
+    
+    private void processComments() {
+        List<String> comments = new ArrayList<>();
+        String sequencedSamples = "";
+        BufferedReader reader = null;
+        try {            
+            reader = new BufferedReader(new FileReader(filename));
+            String line;
+            while((line = reader.readLine()) != null) {     
+                if (line.startsWith("#sequenced_samples")) {
+                    sequencedSamples = line;
+                }
+                else if (line.startsWith("#")) {
+                    comments.add(line);
+                }
+                else {
+                    // no more comments, go on processing
+                    break;
+                }
+            }            
+            reader.close();
+        }        
+        catch (Exception e) {
+            throw new ItemStreamException(e);
+        }
+        
+        // Add comments to the config for the writer to access later
```

Comment:
I like Angelica's approach - to use the job context not the Batch Configuration to share data across the steps.